MoreLinq.Test/RandomSubsetTest.cs

```diff
@@ -25,42 +25,50 @@ public void TestRandomSubsetIsLazy()
         /// Verify that invoking RandomSubsets on a <c>null</c> sequence results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestRandomSubsetNullSequence()
         {
-            const IEnumerable<int> nullSequence = null;
-            nullSequence.RandomSubset(10);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const IEnumerable<int> nullSequence = null;
+                nullSequence.RandomSubset(10);
+            });
         }
 
         /// <summary>
         /// Verify that invoking RandomSubsets on a <c>null</c> sequence results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestRandomSubsetNullSequence2()
         {
-            const IEnumerable<int> nullSequence = null;
-            nullSequence.RandomSubset(10, new Random());
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const IEnumerable<int> nullSequence = null;
+                nullSequence.RandomSubset(10, new Random());
+            });
         }
 
         /// <summary>
         /// Verify that involving RandomSubsets with a subset size less than 0 results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestRandomSubsetNegativeSubsetSize()
         {
-            Enumerable.Range(1, 10).RandomSubset(-5);
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                Enumerable.Range(1, 10).RandomSubset(-5);
+            });
         }
 
         /// <summary>
         /// Verify that involving RandomSubsets with a subset size less than 0 results in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestRandomSubsetNegativeSubsetSize2()
         {
-            Enumerable.Range(1, 10).RandomSubset(-1, new Random());
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                Enumerable.Range(1, 10).RandomSubset(-1, new Random());
```

Comment:
The call to `Range` should go outside the lambda to `Assert.Throws` and the one really under test for an exception, `RandomSubset`, should go inside. Otherwise, there is the risk of a false positive since we don't know if `Range` or `RandomSubset` threw `ArgumentOutOfRangeException`.